src/main/java/app/persistence/AccountMapper.java

```diff
@@ -0,0 +1,83 @@
+package app.persistence;
+
+import app.exceptions.AccountCreationException;
+import app.exceptions.DatabaseException;
+import io.javalin.http.Context;
+
+import java.sql.*;
+import java.util.ArrayList;
+
+public class AccountMapper {
+    public static ArrayList<String> getAllEmailsFromAccount(ConnectionPool connectionPool) throws DatabaseException {
+        ArrayList<String> emails = new ArrayList<>();
+
+        String sql = "select email from account";
+
+        try (Connection connection = connectionPool.getConnection();
+             PreparedStatement ps = connection.prepareStatement(sql)) {
+
+            ResultSet rs = ps.executeQuery();
+            while (rs.next()) {
+                emails.add(rs.getString("email"));
+            }
+            return emails;
+
+        } catch (SQLException e) {
+            throw new DatabaseException("Fejl i at hente emails", "Error in getAllEmailsFromAccount", e.getMessage());
+        }
+    }
+
+    public static int getIdFromAccountEmail(String email, ConnectionPool connectionPool) throws AccountCreationException {
+        int accountId = 0;
+        String sql = "SELECT account_id FROM account WHERE email = ?";
+
+        try (Connection connection = connectionPool.getConnection();
+             PreparedStatement ps = connection.prepareStatement(sql)) {
+
+            ps.setString(1, email);
+            ResultSet rs = ps.executeQuery();
+
+            if (rs.next()) {
+                accountId = rs.getInt("account_id");
+            }
+
+        } catch (SQLException e) {
+            throw new AccountCreationException("Fejl ved søgning efter account ID", "Error in getIdFromAccountEmail" ,e.getMessage());
+        }
+
+        return accountId;
+    }
+
+    public static int createAccount(String name, String adress, int zip, String phone, String email, ConnectionPool connectionPool) throws AccountCreationException {
+        int accountId;
+        String sql = "INSERT INTO account (email, password, name, role, address, zip_code, phone)" +
+                " VALUES (?, ?, ?, ?, ?, ?, ?) ";
```

Comment:
Jeg tror denne metode kan skrives simplere ved at bruge SQL keywordet "RETURNING":
"INSERT INTO account (email, password, name, role, address, zip_code, phone)" VALUES (?, ?, ?, ?, ?, ?, ?) RETURNING account_id". For herved behøver man ikke bruge Statement.RETURN_GENERATED_KEYS.
Prøv at se OrderMapper.createActiveOrder i vores Cupcake projekt.